sdcm/cluster.py

```diff
@@ -2730,6 +2730,7 @@ def get_peers_info(self):
     def get_gossip_info(self):
         gossip_info = self.run_nodetool('gossipinfo', verbose=False, warning_event_on_exception=(Exception,),
                                         publish_event=False)
+        LOGGER.debug("get_gossip_info: %s", gossip_info)
         gossip_node_schemas = {}
         schema = ip = status = dc = ''
```

Comment:
i meant like this:

```suggestion
        for line in gossip_info.stdout.split():
            schema = ip = status = dc = ''
            if line.startswith('SCHEMA:'):
                schema = line.replace('SCHEMA:', '')
            elif line.startswith('RPC_ADDRESS:'):
                ip = line.replace('RPC_ADDRESS:', '')
            elif line.startswith('STATUS:'):
                status = line.replace('STATUS:', '').split(',')[0]
            elif line.startswith('DC:'):
                dc = line.replace('DC:', '').split(',')[0]
            if schema and ip and status:
                if node := self.parent_cluster.find_node_by_ip(ip):
                    gossip_node_schemas[node] = {'schema': schema, 'status': status, 'dc': dc}
                elif status in self.GOSSIP_STATUSES_FILTER_OUT:
                    LOGGER.debug("Get gossip info. Node with IP %s is not found in the cluster. Node status is: %s",
                                 ip, status)
                else:
                    LOGGER.error("Get gossip info. Failed to find a node with status %s in the cluster by IP: %s",
                                 status, ip)
```

Reply:
I understood you, but it's wrong. I don't want to reset all variables every iteration. 
Example of output:
```
/10.0.3.241
  generation:1643553060
  heartbeat:1800
  DC:eu-west
  X6:8
  X7:12
  X9:org.apache.cassandra.locator.Ec2Snitch
  LOAD:902144
  NET_VERSION:0
  RELEASE_VERSION:3.0.8
  RPC_ADDRESS:10.0.3.241
  HOST_ID:8665c6db-5302-4669-8496-0fa5f215a351
  STATUS:LEFT,969007214824697289,1643813164088995150
  X1:ALTERNATOR_STREAMS,ALTERNATOR_TTL,CDC,CDC_GENERATIONS_V2,COMPUTED_COLUMNS,CORRECT_COUNTER_ORDER,CORRECT_IDX_TOKEN_IN_SEC
ONDARY_INDEX,CORRECT_NON_COMPOUND_RANGE_TOMBSTONES,CORRECT_STATIC_COMPACT_IN_MC,COUNTERS,DIGEST_FOR_NULL_VALUES,DIGEST_INSENSITIVE_TO_EXPIRY,DIGEST_MULTIPARTITION_READ,HINTED_HANDOFF_SEPARATE_CONNECTION,INDEXES,
LARGE_PARTITIONS,LA_SSTABLE_FORMAT,LWT,MATERIALIZED_VIEWS,MC_SSTABLE_FORMAT,MD_SSTABLE_FORMAT,NONFROZEN_UDTS,PER_TABLE_CACHING,PER_TABLE_PARTITIONERS,RANGE_SCAN_DATA_VARIANT,RANGE_TOMBSTONES,ROLES,ROW_LEVEL_REPA
IR,SCHEMA_TABLES_V3,SEPARATE_PAGE_SIZE_AND_SAFETY_LIMIT,STREAM_WITH_RPC_STREAM,TOMBSTONE_GC_OPTIONS,TRUNCATION_TABLE,UDA,UNBOUNDED_RANGE_TOMBSTONES,VIEW_VIRTUAL_COLUMNS,WRITE_FAILURE_REPLY,XXHASH
  X3:3
  SCHEMA:7ebb1427-210c-3fd3-914e-98601b00c069
  X2:system_traces.events:0.000000;keyspace1.standard1:0.000000;system_traces.sessions:0.000000;system_distributed.service_le
vels:1.000000;system_distributed.view_build_status:0.000000;system_distributed_everywhere.cdc_generation_descriptions_v2:0.000000;system_auth.role_members:0.000000;system_distributed.cdc_streams_descriptions_v2:
0.000000;system_traces.node_slow_log_time_idx:0.000000;system_traces.node_slow_log:0.000000;system_auth.roles:1.000000;system_distributed.cdc_generation_timestamps:0.000000;system_traces.sessions_time_idx:0.0000
00;system_auth.role_attributes:0.000000;
  RACK:1a
  X5:0:18446744073709551615:1643553147783
  X4:1
  X8:v2;1643553175432;2cffd160-f901-4ca1-97a0-5bfb1fd30a34
```

So the code goes line by line and collects schema & ip & status. Just when ALL values are found and the node is recognized, I can reset them

Reply:
but this is exactly what you are doing in line 2756:
```
...
                else:
                    LOGGER.error("Get gossip info. Failed to find a node with status %s in the cluster by IP: %s",
                                 status, ip)

                schema = ip = status = dc = ''
```

or am i missing something else?

Reply:
```
            if schema and ip and status:
                if node := self.parent_cluster.find_node_by_ip(ip):
                    gossip_node_schemas[node] = {'schema': schema, 'status': status, 'dc': dc}
                elif status in self.GOSSIP_STATUSES_FILTER_OUT:
                    LOGGER.debug("Get gossip info. Node with IP %s is not found in the cluster. Node status is: %s",
                                 ip, status)
                else:
                    LOGGER.error("Get gossip info. Failed to find a node with status %s in the cluster by IP: %s",
                                 status, ip)

                schema = ip = status = dc = ''
```

Only when True `if schema and ip and status`

Reply:
ah, missed that indentation!!